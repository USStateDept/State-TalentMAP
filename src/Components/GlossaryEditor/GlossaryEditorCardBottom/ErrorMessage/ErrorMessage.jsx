import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ showEmptyWarning, showResponseError }) => (
  <span className="usa-input-error-message" role="alert">
    { showEmptyWarning && 'Title and definition cannot be blank.' }
    { showResponseError && 'Error updating term.' }
  </span>
);

ErrorMessage.propTypes = {
  showEmptyWarning: PropTypes.bool,
  showResponseError: PropTypes.bool,
};

ErrorMessage.defaultProps = {
  showEmptyWarning: false,
  showResponseError: false,
};

export default ErrorMessage;
