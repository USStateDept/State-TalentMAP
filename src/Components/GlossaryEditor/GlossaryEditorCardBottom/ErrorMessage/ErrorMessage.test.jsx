import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessageComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<ErrorMessage />);
    expect(wrapper).toBeDefined();
  });

  it('displays an empty string error', () => {
    const wrapper = shallow(<ErrorMessage showEmptyWarning />);
    expect(wrapper.find('.usa-input-error-message').text()).toBe('Title and definition cannot be blank.');
  });

  it('displays a response error', () => {
    const wrapper = shallow(<ErrorMessage showResponseError />);
    expect(wrapper.find('.usa-input-error-message').text()).toBe('Error updating term.');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ErrorMessage />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when showEmptyWarning is true', () => {
    const wrapper = shallow(<ErrorMessage showEmptyWarning />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when showResponseError is true', () => {
    const wrapper = shallow(<ErrorMessage showResponseError />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when showResponseError and showEmptyWarning are true', () => {
    const wrapper = shallow(<ErrorMessage showEmptyWarning showResponseError />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
