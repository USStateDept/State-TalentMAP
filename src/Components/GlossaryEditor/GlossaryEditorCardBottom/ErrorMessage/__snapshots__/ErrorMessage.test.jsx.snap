// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessageComponent matches snapshot 1`] = `null`;

exports[`ErrorMessageComponent matches snapshot when showEmptyWarning is true 1`] = `
<span
  className="usa-input-error-message"
  role="alert"
>
  Title and definition cannot be blank.
</span>
`;

exports[`ErrorMessageComponent matches snapshot when showResponseError and showEmptyWarning are true 1`] = `
<span
  className="usa-input-error-message"
  role="alert"
>
  Title and definition cannot be blank.
</span>
`;

exports[`ErrorMessageComponent matches snapshot when showResponseError is true 1`] = `
<span
  className="usa-input-error-message"
  role="alert"
>
  error
</span>
`;
